src/components/Settings/Speech/Speech.container.js

```diff
@@ -73,14 +87,16 @@ export class SpeechContainer extends Component {
     this.updateSettingsTimeout = setTimeout(async () => {
       const {
         speech: {
-          options: { voiceURI, pitch, rate }
+          options: { voiceURI, pitch, rate },
+          elevenLabsApiKey
         }
       } = this.props;
 
       const speech = {
         voiceURI,
         pitch,
         rate,
+        elevenLabsApiKey,
```

Comment:
same comment than above

Reply:
Agus you can keep accessing the prop as you are doing now, or use `this.props.elevenLabsApiKey` as you suggested.  

Martin, you are correct that the prop comes in through the params. However, it’s important to note that **all speech-related fields must be taken from the props and passed into the `updateSettings` function**.  

The reason is that the server expects the full set of speech settings. If any field is missing, the server will erase it from the stored configuration. This behavior comes directly from how the function is implemented on the server side:  

https://github.com/cboard-org/cboard-api/blob/21d4f7e26a284943370cbdb1ff35d0785c9ebe79/api/controllers/settings.js#L8-L33